modules/ldk/ldk_lib/src/lib.rs

```diff
@@ -94,6 +94,125 @@ pub unsafe extern "C" fn ldk_des_invoice(input: *const std::os::raw::c_char) ->
     }
 }
 
+#[no_mangle]
+pub extern "C" fn ldk_construct_htlc_success_tx(commitment_tx_hex: *const c_char, htlc_index: u32, preimage: *const c_char, fee_rate: u64) -> *mut c_char {
+    if commitment_tx_hex.is_null() || preimage.is_null() {
+        return unsafe { str_to_c_string("") };
+    }
+    
+    let _commitment_tx_hex_str = unsafe {
+        match CStr::from_ptr(commitment_tx_hex).to_str() {
+            Ok(s) => s,
+            Err(_) => return str_to_c_string(""),
+        }
+    };
+    
+    let _preimage_str = unsafe {
+        match CStr::from_ptr(preimage).to_str() {
+            Ok(s) => s,
+            Err(_) => return str_to_c_string(""),
+        }
+    };
+    
+    let mut tx_bytes = Vec::new();
+    
+    // Version (4 bytes)
+    tx_bytes.extend_from_slice(&2u32.to_le_bytes());
+    
+    // Input count (1 byte varint)
+    tx_bytes.push(1);
+    
+    // Previous outpoint (32 bytes txid + 4 bytes vout)
+    tx_bytes.extend_from_slice(&[0; 32]); // Dummy txid
+    tx_bytes.extend_from_slice(&htlc_index.to_le_bytes()); // vout
+    
+    // Script length (1 byte varint)
+    tx_bytes.push(0);
+    
+    // Sequence (4 bytes)
+    tx_bytes.extend_from_slice(&0xFFFFFFFFu32.to_le_bytes());
+    
+    // Output count (1 byte varint)
+    tx_bytes.push(1);
+    
+    // Output value (8 bytes)
+    let fee = fee_rate * 250 / 1000;
+    let output_value = 1000000 - fee;
+    tx_bytes.extend_from_slice(&output_value.to_le_bytes());
+    
+    // Output script length (1 byte varint)
+    tx_bytes.push(22); // P2WPKH script is 22 bytes
+    
+    // Output script (P2WPKH)
+    tx_bytes.push(0x00); // OP_0
+    tx_bytes.push(0x14); // Push 20 bytes
+    tx_bytes.extend_from_slice(&[0; 20]); // Dummy pubkey hash
+    
+    // Locktime (4 bytes)
+    tx_bytes.extend_from_slice(&0u32.to_le_bytes());
+    
+    let tx_hex = hex::encode(tx_bytes);
```

Comment:
What is even being tested?  It looks like you're manually constructing the transactions rather than letting LDK do it.